orchagent/switchorch.cpp

```diff
@@ -33,17 +33,100 @@ const map<string, sai_packet_action_t> packet_action_map =
+        m_sensorsPollerTimer (new SelectableTimer((timespec { .tv_sec = DEFAULT_ASIC_SENSORS_POLLER_INTERVAL, .tv_nsec = 0 })))
 {
     m_restartCheckNotificationConsumer = new NotificationConsumer(db, "RESTARTCHECK");
     auto restartCheckNotifier = new Notifier(m_restartCheckNotificationConsumer, this, "RESTARTCHECK");
     Orch::addExecutor(restartCheckNotifier);
+
+    initSensorsTable();
+    auto executorT = new ExecutableTimer(m_sensorsPollerTimer, this, "ASIC_SENSORS_POLL_TIMER");
+    Orch::addExecutor(executorT);
 }
 
-void SwitchOrch::doTask(Consumer &consumer)
+void SwitchOrch::doCfgSensorsTableTask(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
+        string table_attr = kfvKey(t);
+        string op = kfvOp(t);
+
+        if (op == SET_COMMAND)
+        {
+            FieldValueTuple fvt = kfvFieldsValues(t)[0];
+            SWSS_LOG_NOTICE("ASIC sensors : set %s(%s) to %s", table_attr.c_str(), fvField(fvt).c_str(), fvValue(fvt).c_str());
+
+            if (table_attr == ASIC_SENSORS_POLLER_STATUS)
+            {
+                if (fvField(fvt) == "admin_status")
+                {
+                    if (fvValue(fvt) == "enable" && !m_sensorsPollerEnabled)
+                    {
+                        m_sensorsPollerTimer->start();
+                        m_sensorsPollerEnabled = true;
+                    }
+                    else if (fvValue(fvt) == "disable")
+                    {
+                        m_sensorsPollerEnabled = false;
+                    }
+                    else
+                    {
+                        SWSS_LOG_ERROR("ASIC sensors : unsupported operation for poller state %d",m_sensorsPollerEnabled);
+                    }
+                }
+                else
+                {
+                    SWSS_LOG_ERROR("ASIC sensors : unsupported field");
+                }
+            }
+            else if (table_attr == ASIC_SENSORS_POLLER_INTERVAL)
+            {
+                uint32_t interval=to_uint<uint32_t>(fvValue(fvt));
+
+                if (fvField(fvt) == "interval")
+                {
+                    if (interval != m_sensorsPollerInterval)
+                    {
+                        auto intervT = timespec { .tv_sec = interval , .tv_nsec = 0 };
+                        m_sensorsPollerTimer->setInterval(intervT);
+                        m_sensorsPollerInterval = interval;
+                        m_sensorsPollerIntervalChanged = true;
+                    }
+                    else
+                    {
+                        SWSS_LOG_INFO("ASIC sensors : poller interval unchanged : %d seconds",m_sensorsPollerInterval);
+                    }
+                }
+                else
+                {
+                    SWSS_LOG_ERROR("ASIC sensors : unsupported field");
```

Comment:
should you log also on which component this field is not supported ?, same on 117

Reply:
will add the changes